File: Goods.Social.Network/Goods.System.Social.Network.Microservice.Posts/Entities/DTO/PostDTO.cs
namespace Goods.System.Social.Network.Microservice.Posts.Entities.DTO
{
    /// <summary>
    /// Пост DTO
    /// </summary>
    public class PostDTO
    {
        /// <summary>
        /// Id поста
        /// </summary>
        public int Id { get; set; }
        /// <summary>
        /// Путь до фотографии
        /// </summary>
        public string Image { get; set; } = string.Empty;
        /// <summary>
        /// Заголовок
        /// </summary>
        public string Title { get; set; } = string.Empty;
        /// <summary>
        /// Дата создания
        /// </summary>
        public DateTime DateCreate { get; set; }
        /// <summary>
        /// Текст коммента
        /// </summary>
        public int UserId { get; set; }
        /// <summary>
        /// Имя автора
        /// </summary>
        public string FirstNameUser { get; set; } = string.Empty;
        /// <summary>
        /// Фамилия автора
        /// </summary>
        public string LastNameUser { get; set; } = string.Empty;
        /// <summary>
        /// Текст поста
        /// </summary>
        public string Text { get; set; } = string.Empty;
        /// <summary>
        /// Теги
        /// </summary>
        public string Tags { get; set; } = string.Empty;
        /// <summary>
        /// Рейтинг поста
        /// </summary>
        public int Rating { get; set; }
        /// <summary>
        /// Тип реакции, которую поставил твой юзер
        /// </summary>
        public string TypeReaction { get; set; } = string.Empty;
    }
}

File: web-app/libs/data/src/controllers/post/post-dto.ts
/** GET /api/v1/post/{id} Return */
export type PostGetReturn = {
  id: number;
  /** Путь до фотографии */
  image: string | null;
  title: string | null;
  dateCreate: Date | string;
  userId: number;
  firstNameUser: string | null;
  lastNameUser: string | null;
  text: string | null;
  tags: string | null;
  rating: number;
  typeReaction: string | null;
};

/** DELETE /api/v1/post/{id} Query */
export type PostDeleteQuery = {
  userId: number;
};

/** GET /api/v1/post Query */
export type PostGetListQuery = {
  search: string;
  userId: number;
  number: number;
};

/** GET /api/v1/post Return */
export type PostGetListReturn = {
  countAllPosts: number;
  pageCount: number;
  numberPage: number;
  posts: PostGetReturn[] | null;
};

/** POST /api/v1/post Body */
export type PostPostBody = {
  image: File;
  title: string;
  dateCreate: Date;
  userId: number;
  text: string;
  tags: string;
};

/** PUT /api/v1/post Body */
export type PostPutBody = {
  id: number;
  image?: File;
  title: string;
  userId: number;
  text: string;
  tags: string;
};
